protocols/relay/src/v2/client/transport.rs

```diff
@@ -217,6 +225,35 @@ impl Transport for ClientTransport {
     fn address_translation(&self, _server: &Multiaddr, _observed: &Multiaddr) -> Option<Multiaddr> {
         None
     }
+
+    fn poll(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<TransportEvent<Self::ListenerUpgrade, Self::Error>>
+    where
+        Self: Sized,
+    {
+        loop {
+            if !self.pending_to_behaviour.is_empty() {
+                match self.to_behaviour.poll_ready(cx) {
+                    Poll::Ready(Ok(())) => {
+                        let msg = self
+                            .pending_to_behaviour
+                            .pop_front()
+                            .expect("Called !is_empty().");
+                        let _ = self.to_behaviour.start_send(msg);
+                        continue;
+                    }
+                    Poll::Ready(Err(_)) => unreachable!("Receiver is never dropped."),
+                    Poll::Pending => {}
+                }
+            }
+            match self.listeners.poll_next_unpin(cx) {
+                Poll::Ready(Some(event)) => return Poll::Ready(event),
+                _ => return Poll::Pending,
```

Comment:
This also returns Poll::Pending in case of `Poll::Ready(None)`, in which case no waker is informed.
`Poll::Ready(None)` will only happen if we currently don't have a listener. If we add a new listener again with `listen_on`, I would assume we poll the transport afterwards again, thus registering the waker with the new listener.  Thus I think it's okay to do this here.

Reply:
The alternative would be to store the waker and call it once we add a new listener. 
Don't have a strong opinion here; happy to change it if you prefer the latter.

Reply:
This is not the only place which we will have to consider in case we ever do the advanced polling. Thus, I think we can keep this as is for now, being consistent with the remaining code base. Thanks @elenaf9!